src/canvas/SelectableCanvas.ts

```diff
@@ -860,15 +860,27 @@ export class SelectableCanvas<EventSpec extends CanvasEvents = CanvasEvents>
     pointer: Point
   ): FabricObject | undefined {
     const target = this._searchPossibleTargets(objects, pointer);
-    // if we found something in this.targets, and the group is interactive, return that subTarget
+
+    // if we found something in this.targets, and the group is interactive, return the innermost subTarget
+    // that is still interactive
     // TODO: reverify why interactive. the target should be returned always, but selected only
     // if interactive.
-    return target &&
+    if (
+      target &&
       isCollection(target) &&
       target.interactive &&
       this.targets[0]
-      ? this.targets[0]
-      : target;
+    ) {
+      /** {@link this.targets} is ordered bottom-up */
+      let subTarget = this.targets[0];
+      while (subTarget.parent && !subTarget.parent.interactive) {
+        subTarget = subTarget.parent;
+      }
+
+      return subTarget;
```

Comment:
I understand you don't want to break anything but... this feels a bit too much. Seems that `targets` is not fitting enough.
Also I am not 100% sure this is what is expected for all cases.
Consider 2 siblings under an interactive group. A is an interactive group itself, B is a non interactive group. B is on top. Both have overlapping children A1, B1 at some point P.
Let's say we are searching targets on P. Who should be returned, A1 or B?
This code returns B.
I think that is correct.

Reply:
But what if B's onSelect returns true, signaling to prevent the selection?
Then I think A1 should be returned. This code is not doing that.

Reply:
I am reminding the PR that refactors this method and returns an array without populating `targets`. I understand we want v6 out. However I am surfacing it for consideration.

Reply:
Perhaps a sound/robust method would return 2 arrays: subTargets, selectionTargets

Reply:
And then we should define strictly that a subTarget is not a selection target

Reply:
> But what if B's onSelect returns true, signaling to prevent the selection?
Then I think A1 should be returned. This code is not doing that.

I don't agree. `onSelect` only determines if the object should be selected once the decision is already finalised, it doesn't determine if the object should be picked as potential candidate as target. That is decided in `_searchPossibleTargets` and `_checkTarget`.

I agree that `targets` is not perfect. I haven't even opened an issue about the order in/out events are fired (bottom-up in fabric, wrong IMO) because that's more subject to discussion. But indeed those bigger changes are for after v6 since are more significant and require an in-depth discussion.

This PR instead just focuses on fixing a behaviour as intended in current beta without any breaking change.

Reply:
What if you use
>  this.targets.find(target => target.parent?.interactive) ?? target

I think it is more correct

Reply:
And please use something different than bottom up which is very confusing IMO.
Front to back maybe?

Reply:
I like bottom-up, I find it more clear than "front-to-back" which sounds more about zIndex, i.e. stack between siblings. What do you prefer @asturur ?

Reply:
both bottom up and front-back are confusing to me, i was indeed trying to understand what does it mean. Does it mean from inner to outer?

Reply:
I think we should use descendant-ancestor
Anything else is open to confusion

Reply:
> What if you use
> 
> > this.targets.find(target => target.parent?.interactive) ?? target
> 
> I think it is more correct

What about this? I think it is more straight forward

Reply:
> I think we should use descendant-ancestor

👍 

> this.targets.find(target => target.parent?.interactive) ?? target

👍

Reply:
mmmm this code here is tricky.
It opens to more edge cases.
This will stop at finding the fist one that has an interactive parent.
It doesn't tell us anything of all the other outer objects. Unless i m misreading it.
```
this.targets.find(target => target.parent?.interactive) ?? target
```
I think you have to start from the outermost and dig in by interactivity, i see no other way if not redo the same line of checks the original code was doing

Reply:
I also think current code is not correct.
You have to cycle from the outermost and check. let me try to suggest